command/agent/local.go

```diff
@@ -316,6 +316,13 @@ func (l *localState) setSyncState() error {
 	l.Lock()
 	defer l.Unlock()
 
+	for id, _ := range l.services {
+		// If the local service doesn't exist remotely, then sync it
+		if _, ok := services.Services[id]; !ok {
+			l.serviceStatus[id] = syncStatus{inSync: false}
+		}
+	}
+
 	if services != nil {
```

Comment:
It looks like we guard against services result being nil. Can we move that above the new code and just do something like:

```
if services == nil {
 services = make(map...)
}
```
This way we can avoid the guard logic and nesting. I think the new code here may be susceptible to a nil pointer dereference in those cases. Not that I see how we could get nil back.